h2o-py/h2o/estimators/infogram.py

```diff
@@ -0,0 +1,1037 @@
+                if len(infogram_algorithm_params_dict[k]) == 1: #single parameter
+                    infogram_algorithm_params_dict[k] = infogram_algorithm_params_dict[k][0]
+            return infogram_algorithm_params_dict
+        else:
+            return self._parms.get("infogram_algorithm_params")
+
+    @infogram_algorithm_params.setter
+    def infogram_algorithm_params(self, infogram_algorithm_params):
+        assert_is_type(infogram_algorithm_params, None, dict)
+        if infogram_algorithm_params is not None and infogram_algorithm_params != "":
+            for k in infogram_algorithm_params:
+                if ("[" and "]") not in str(infogram_algorithm_params[k]):
+                    infogram_algorithm_params[k] = [infogram_algorithm_params[k]]
+            self._parms["infogram_algorithm_params"] = str(json.dumps(infogram_algorithm_params))
+        else:
+            self._parms["infogram_algorithm_params"] = None
+
+    @property
+    def protected_columns(self):
+        """
+        predictors that are to be excluded from model due to them being discriminatory or inappropriate for whatever
+        reason.
+
+        Type: ``List[str]``.
+        """
+        return self._parms.get("protected_columns")
+
+    @protected_columns.setter
+    def protected_columns(self, protected_columns):
+        assert_is_type(protected_columns, None, [str])
+        self._parms["protected_columns"] = protected_columns
+
+    @property
+    def net_information_threshold(self):
+        """
+        conditional information for core infogram threshold between 0 and 1 that is used to decide whether a predictor's
+        conditional information is high enough to be chosen into the admissible feature set.  If not set, will default
+        to 0.1.
+
+        Type: ``float``, defaults to ``4.940656458e-324``.
+        """
+        return self._parms.get("net_information_threshold")
+
+    @net_information_threshold.setter
+    def net_information_threshold(self, net_information_threshold):
+        if net_information_threshold <= 4.940656458e-324: # not set
+            if self._parms["protected_columns"] is None:
+                self._parms["net_information_threshold"]=0.1
+        else:  # set
+            if self._parms["protected_columns"] is not None: # fair infogram
+                warnings.warn("Should not set net_information_threshold for fair infogram runs.  Set safety_index_threshold instead.  Using default of 0.1 if not set", RuntimeWarning)
+            else:
+                self._parms["net_information_threshold"]=net_information_threshold
+
+    @property
+    def total_information_threshold(self):
+        """
+        relevance threshold for core infogram between 0 and 1 that is used to decide whether a predictor's relevance
+        level is high enough to be chosen into the admissible feature set.  If not set, will default to 0.1.
+
+        Type: ``float``, defaults to ``4.940656458e-324``.
```

Comment:
I don't think it is a good idea to have some complicated magic float number as a default especially if we say the default is 0.1 and on the next line we say it's ~5e-324.

In cases where the value is supposed to be positive (or even limited between 0 and 1), I'd use either the default (here 0.1) or -1  since we use it already as the "AUTO" value.

Reply:
@tomasfryda This is a known issue. I reported it to @wendycwong but it has not been fixed yet (she's working through higher priority items first).  I am not sure why it's being inserted into the clients as the default (when it's not the correct value).  Does anyone know where (I think in Java) this is coming from?  If we can find the source, we can fix it... I was not able to figure out why/where it's being added (in both R and Python client).

Reply:
@wendycwong said this is fixed on the original PR & CV branch.